elixir/apps/domain/lib/domain/billing.ex

```diff
@@ -36,17 +36,33 @@ defmodule Domain.Billing do
 
   # Limits and Features
 
+  def users_limit_exceeded?(%Accounts.Account{} = account, users_count) do
+    not is_nil(account.limits.users_count) and
+      users_count > account.limits.users_count
```

Comment:
Just wondering for my own edification if the `not is_nil(account.limits.users_count)` is needed? 

When I tried in IEx real quick, it looks like Elixir allows greater than / less than comparisons against `nil`:
```
iex(8)> 0 > nil
false
```
But, I may be overlooking something.

Reply:
We should not rely on comparison between different types, `nil` for example is just an atom `:nil` and can lead to all sorts of weird behaviors:
```
iex> nil >= 0
true

iex> nil >= DateTime.utc_now()
false
```

For the same reason, we should not compare two DateTimes directly (`dt1 > dt2`) and must use `DateTime.compate/2` instead. See https://hexdocs.pm/elixir/1.12.3/Kernel.html#module-structural-comparison

Reply:
Gotcha.  Appreciate the explanation.